src/index.tsx

```diff
@@ -383,6 +389,20 @@ export function useMicrophone() {
     setIsMicrophoneOn(MembraneWebRTCModule.isMicrophoneOn);
     return () => eventListener.remove();
   }, []);
+  /**
+   * This sets the voice activity detection state.
+   */
+  useEffect(() => {
+    if (isVADOn) {
+      RNSoundLevel.onNewFrame = (data: VADData) => {
```

Comment:
why are we doing it in useEffect, if we do it only when vad is started and there is no cleanup?
I see two options to do it better:
1. Make only `startVADMonitor` and `stopVADMonitor`. `startVADMonitor` sets the `onNewFrame` callback and calls `RNSoundLevel.start`. User is responsible for calling `stopVADMonitor` when they're done with it, but they have more control over it and it's simpler. Also you can get rid of `isVADOn` state.
2. Create a separate hook with `useEffect` (for example `useVADMonitor`). When used, the hook starts up RNSoundLevel and sets up callback (both in useEffect). When no longer used, it cleans up automatically (in `useEffect` return function). It's easier to use, but user has less control (what happens when they use the hook in two different places in the app?).